### PR TITLE
feat(io-engine/api): add snapshot rebuild api

### PR DESCRIPTION
**Pull request by tiagolobocastro**

### CODE REVIEW

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+service SnapshotRebuildRpc {
+  rpc CreateSnapshotRebuild(CreateSnapshotRebuildRequest) returns (SnapshotRebuild) {}
+  rpc ListSnapshotRebuild(ListSnapshotRebuildRequest) returns (ListSnapshotRebuildResponse) {}
+  rpc DestroySnapshotRebuild(DestroySnapshotRebuildRequest) returns (google.protobuf.Empty) {}
+}
+
+message CreateSnapshotRebuildRequest {
+  // The uuid of the rebuilding replica.
+  string                     replica_uuid = 1;
```
Comment (niladrih) [inline apis/io-engine/protobuf/v1/snapshot-rebuild.proto:17]: nit: is this indentation style standard? Looks kind of odd.

**Review by niladrih — APPROVED** [review]
Left one nitpick comment. LGTM.

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto (hunk 1 above)
Comment (tiagolobocastro) [reply apis/io-engine/protobuf/v1/snapshot-rebuild.proto:17]: Don't think we have a standard for proto, maybe something to look into.
I tend to do it this way, example: https://github.com/openebs/mayastor-dependencies/blob/d9b52299f1c837504296cf35b4790d3f9dcf271a/apis/io-engine/protobuf/v1/test.proto#L41

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  // In case of crash/restart, we can first
+  // 1. read allocated data bitmap
+  // and then depending on the resume flag:
+  // true:
+  // 2. read data starting from the last known checkpoint
+  // false:
+  // 2. start again from scratch
+  bool                             resume = 3;
+  // A bitmap where set bits indicate which
+  // segment of the snapshot are allocated and thefore
```
Comment (Abhinandan-Purkait) [inline apis/io-engine/protobuf/v1/snapshot-rebuild.proto]: thefore -> therefore?

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  // List all rebuilds or just the rebuild pertaining to the given replica uuid.
+  optional string replica_uuid = 1;
+}
+message ListSnapshotRebuildResponse {
+  repeated SnapshotRebuild rebuilds = 1;
+}
+
+message DestroySnapshotRebuildRequest {
+  string replica_uuid = 1;
+}
```
Comment (Abhinandan-Purkait) [inline apis/io-engine/protobuf/v1/snapshot-rebuild.proto]: Blank line needed at the EOF?

**Review by Abhinandan-Purkait — APPROVED** [review]

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto (hunk 2 above)
Comment (tiagolobocastro) [reply apis/io-engine/protobuf/v1/snapshot-rebuild.proto]: doh, fixed thanks

File: apis/io-engine/protobuf/v1/snapshot-rebuild.proto (hunk 3 above)
Comment (tiagolobocastro) [reply apis/io-engine/protobuf/v1/snapshot-rebuild.proto]: Done

**Pull request merged**